src/main/java/com/newrelic/jfr/toevent/MethodSampleMapper.java

```diff
@@ -25,6 +25,12 @@
   public static final String EVENT_NAME = "jdk.ExecutionSample";
   public static final String NATIVE_EVENT_NAME = "jdk.NativeMethodSample";
 
+  private final String eventName;
+
+  public MethodSampleMapper(final String eventName) {
```

Comment:
I totally grok adding the field and being able to reuse.  The risk is that a user could pass in any old thing.  How about making the constructor private and building 2 static factory methods, one for each supported event type?  Then users can simply `MethodSampleMapper.forExecutionSample()` or `MethodSampleMapper.forNativeMethodSample()`.

Reply:
Fixed.